src/components/SingleArticle.jsx

```diff
@@ -1,17 +1,23 @@
 import React, { useEffect, useState } from "react";
 import { useParams } from "react-router-dom";
-import { getArticleById } from "../api";
+import { getArticleById, getCommentsByArticleId } from "../api";
+import CommentCard from "./CommentCard";
 
 const SingleArticle = () => {
   const { article_id } = useParams();
   const [article, setArticle] = useState(null);
+  const [comments, setComments] = useState([]);
   const [isLoading, setIsLoading] = useState(true);
   const [isError, setIsError] = useState(false);
 
   useEffect(() => {
-    getArticleById(article_id)
-      .then(({article}) => {
-        setArticle(article);
+    Promise.all([
+      getArticleById(article_id),
+      getCommentsByArticleId(article_id)
```

Comment:
Rather than doing a request for both, consider making the API call in a different component, such as "CommentList" that then feeds to "CommentCard". This way, you don't have to wait for both requests to render both Components, you could render the Article, then while that is showing, render the comments.